addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,292 @@
+    purchase_ids = fields.One2many('purchase.order', 'requisition_id', string='Purchase Orders', states={'done': [('readonly', True)]})
+    po_line_ids = fields.One2many('purchase.order.line', compute=_compute_get_po_line, string='Products by vendor')
+    line_ids = fields.One2many('purchase.requisition.line', 'requisition_id', string='Products to Purchase', states={'done': [('readonly', True)]}, copy=True)
+    procurement_id = fields.Many2one('procurement.order', string='Procurement', ondelete='set null', copy=False)
+    warehouse_id = fields.Many2one('stock.warehouse', string='Warehouse')
+    state = fields.Selection([('draft', 'Draft'), ('in_progress', 'Confirmed'),
+                            ('open', 'Bid Selection'), ('done', 'PO Created'),
+                            ('cancel', 'Cancelled')], string='Status', track_visibility='onchange', required=True, copy=False, default='draft')
+    multiple_rfq_per_supplier = fields.Boolean(string='Multiple RFQ per vendor')
+    account_analytic_id = fields.Many2one('account.analytic.account', string='Analytic Account')
+    picking_type_id = fields.Many2one('stock.picking.type', string='Picking Type', required=True, default=_default_get_picking_in)
+
+    @api.multi
+    def tender_cancel(self):
+        # try to set all associated quotations to cancel state
+        for tender in self:
+            for purchase_order in tender.purchase_ids:
+                purchase_order.button_cancel()
+                purchase_order.message_post(body=_('Cancelled by the tender associated to this quotation.'))
+        return self.write({'state': 'cancel'})
+
+    @api.multi
+    def tender_in_progress(self):
+        if not all(obj.line_ids for obj in self):
+            raise UserError(_('You can not confirm call because there is no product line.'))
+        return self.write({'state': 'in_progress'})
+
+    @api.multi
+    def tender_open(self):
+        return self.write({'state': 'open'})
+
+    @api.multi
+    def tender_reset(self):
+        self.write({'state': 'draft'})
+        for p_id in self:
+            # Deleting the existing instance of workflow for PO
+            p_id.delete_workflow()
+            p_id.create_workflow()
+        return True
+
+    @api.multi
+    def tender_done(self):
+        return self.write({'state': 'done'})
+
+    @api.multi
+    def open_product_line(self):
+        """ This opens product line view to view all lines from the different quotations, groupby default by product and partner to show comparaison
+            between vendor price
+            @return: the product line tree view
+        """
+        res = self.env['ir.actions.act_window'].for_xml_id('purchase_requisition', 'purchase_line_tree')
+        res['context'] = self.env.context
+        res['context'] = {
+            'search_default_groupby_product': True,
+            'search_default_hide_cancelled': True,
+            'tender_id': self[0].id,
```

Comment:
Not a good idea. Use self.ensure_one() to verify that the self contains only one record.